### PR TITLE
docs: add contract guide

### PR DESCRIPTION
**Pull request by dvilelaf**
## Proposed changes

Add a guide explaining how to implement a contract package and call it.
Depends on #262 

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/creating-contracts.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+2. Initialize the IPFS registry:
+```bash
+autonomy init --reset --author john_doe --remote --ipfs --ipfs-node "/dns/registry.autonolas.tech/tcp/443/https"
+```
+
+3. Create an agent and scaffold the ERC20 contract:
+```bash
+aea create my_aea && cd my_aea
+aea scaffold --to-local-registry contract ERC20Contract /path/to/IERC20.json
```
Comment (angrybayblade) [inline docs/creating-contracts.md:25]: I don't think you need to create the agent to scaffold the contract if you're using `--to-local-registry` flag

Comment (dvilelaf) [reply docs/creating-contracts.md]: Nice!

File: docs/creating-contracts.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    ) -> Optional[JSONLike]:
+        """Gets an account's balance."""
+        contract_instance = cls.get_instance(ledger_api, contract_address)
+
+        return ledger_api.contract_method_call(
+            contract_instance=contract_instance,
+            method_name="balanceOf",
+            owner=owner_address,
+        )
+```
```
Comment (DavidMinarsch) [inline docs/creating-contracts.md:48]: This neatly demonstrates why the PR #262 does not achieve it's purpose. The ledger connection can automatically resolve messages specifying the target `balanceOf` to this method call. No single line of code has to be written by the dev.

Comment (dvilelaf) [reply docs/creating-contracts.md:48]: Yes, but at some point devs might need to have custom code in their implementation, so I think we should leave this and also add the new option.

File: docs/creating-contracts.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+3. Scaffold the ERC20 contract:
+```bash
+aea scaffold --to-local-registry contract ERC20Contract /path/to/IERC20.json
+```
+You'll find the contract in the local registry at `packages/john_doe/contracts/ERC20Contract`.
+
+
+4. Now it is time to call the new method from an agent. Let's say that we would like to get the WETH balance for the WETH account itself. In any skill's `behaviour.py` file, first import the contract package and set the target address:
```
Comment (DavidMinarsch) [inline docs/creating-contracts.md:29]:
```suggestion
4. Now it is time to call the new method from s skill. Let's say that we would like to get the WETH balance for the WETH account itself. In any skill's `behaviour.py` file, first import the contract package and set the target address:
```

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**